### DISCUSSION TITLE
How realistic is it for hono client components to be a fresh competitor?

### DISCUSSION THREAD

**Discussion by tclain** [honojs/hono#2195] [General]
Hi!

Big fan of hono over there and the move to support client components is genius.

Fresh (deno) has an interesting approach where they use preact officially, opening up the whole ecosystem.

I know that client components emulate the react API.

If hono can support any compatibility with preact / or react, it would be game over for so many use cases we could have a truly runtime independent, SSR-first full stack web server. 

how realistic and importantly how can we help to make it happen.

Thanks for the great work. this is awesome!

Tim.

**Comment by yusukebe** [comment]
Hi @tclain !

On the server side, you can render pages using React's `renderToReadableStream()` and so on.

HonoX allows Bring Your Own Renderer, so you can use React or Preact.

https://github.com/honojs/honox?tab=readme-ov-file#byor---bring-your-own-renderer

There are some bugs in HonoX and not enough documentation, so we will be working on improving it.
